src/UseCases/ApplyForMembership/ApplyForMembershipUseCase.php

```diff
@@ -13,44 +13,29 @@
 use WMDE\Fundraising\MembershipContext\Infrastructure\TemplateMailerInterface;
 use WMDE\Fundraising\MembershipContext\Tracking\ApplicationPiwikTracker;
 use WMDE\Fundraising\MembershipContext\Tracking\ApplicationTracker;
-use WMDE\Fundraising\PaymentContext\Domain\Model\PaymentMethod;
 use WMDE\Fundraising\PaymentContext\Domain\PaymentDelayCalculator;
+use WMDE\Fundraising\PaymentContext\UseCases\CreatePayment\CreatePaymentUseCase;
+use WMDE\Fundraising\PaymentContext\UseCases\CreatePayment\FailureResponse;
+use WMDE\Fundraising\PaymentContext\UseCases\CreatePayment\PaymentCreationRequest;
 
-/**
- * @license GPL-2.0-or-later
- */
 class ApplyForMembershipUseCase {
 
-	private ApplicationRepository $repository;
-	private ApplicationTokenFetcher $tokenFetcher;
-	private TemplateMailerInterface $mailer;
-	private MembershipApplicationValidator $validator;
-	private ApplyForMembershipPolicyValidator $policyValidator;
-	/**
-	 * @var ApplicationTracker
-	 * @deprecated See https://phabricator.wikimedia.org/T197112
-	 */
-	private ApplicationTracker $membershipApplicationTracker;
-	private ApplicationPiwikTracker $piwikTracker;
-	private PaymentDelayCalculator $paymentDelayCalculator;
-	private EventEmitter $eventEmitter;
-	private IncentiveFinder $incentiveFinder;
-
-	public function __construct( ApplicationRepository $repository,
-		ApplicationTokenFetcher $tokenFetcher, TemplateMailerInterface $mailer,
-		MembershipApplicationValidator $validator, ApplyForMembershipPolicyValidator $policyValidator,
-		ApplicationTracker $tracker, ApplicationPiwikTracker $piwikTracker,
-		PaymentDelayCalculator $paymentDelayCalculator, EventEmitter $eventEmitter, IncentiveFinder $incentiveFinder ) {
-		$this->repository = $repository;
-		$this->tokenFetcher = $tokenFetcher;
-		$this->mailer = $mailer;
-		$this->validator = $validator;
-		$this->policyValidator = $policyValidator;
-		$this->membershipApplicationTracker = $tracker;
-		$this->piwikTracker = $piwikTracker;
-		$this->paymentDelayCalculator = $paymentDelayCalculator;
-		$this->eventEmitter = $eventEmitter;
-		$this->incentiveFinder = $incentiveFinder;
+	public function __construct(
+		private ApplicationRepository $repository,
+		private ApplicationTokenFetcher $tokenFetcher,
+		private TemplateMailerInterface $mailer,
+		private MembershipApplicationValidator $validator,
+		private ApplyForMembershipPolicyValidator $policyValidator,
+		/**
+		 * @var ApplicationTracker
+		 * @deprecated See https://phabricator.wikimedia.org/T197112
```

Comment:
Great job catching this!

I suggest you leave it in until the CI works and you have all the new tests, then remove it, so we won't have another release that breaks backwards compatibility.